src/unix/darwin.c

```diff
@@ -102,27 +103,39 @@ int uv_exepath(char* buffer, size_t* size) {
 
 
 uint64_t uv_get_free_memory(void) {
-  vm_statistics_data_t info;
+  vm_statistics64_data_t info;
   mach_msg_type_number_t count = sizeof(info) / sizeof(integer_t);
+  uint64_t used;
+  uint64_t not_used;
+  uint64_t available;
 
-  if (host_statistics(mach_host_self(), HOST_VM_INFO,
-                      (host_info_t)&info, &count) != KERN_SUCCESS) {
+  if (host_statistics64(mach_host_self(), HOST_VM_INFO,
+                        (host_info64_t)&info, &count) != KERN_SUCCESS) {
     return 0;
   }
 
-  return (uint64_t) info.free_count * sysconf(_SC_PAGESIZE);
+  used = info.active_count;
+  used += info.inactive_count;
+  used += info.compressor_page_count;
+  used += info.speculative_count;
+  used += info.wire_count;
+  not_used = info.purgeable_count + info.external_page_count;
+  available = uv_get_total_memory() - (used - not_used) * sysconf(_SC_PAGESIZE);
```

Comment:
What if `used < not_used`? That'll underflow and wrap around.

Reply:
Like `available = (used - not_used) * sysconf(_SC_PAGESIZE);`?

Maybe @vtjnash could help us a bit on this?

Reply:
It seems like it would be more direct to add them together `info.free_count + info.purgeable_count + info.external_page_count`. Though I am not sure how justifiable it is to increase free_count by those other values.